### PR TITLE
#537 add h2o learner

### PR DESCRIPTION
**Pull request by Coorsaa**
suggestion for NEWS 
````
new learners
-- classif.h2o.deeplearning
-- classif.h2o.gbm
-- classif.h2o.glm
-- classif.h2o.randomForest
-- regr.h2o.deeplearning
-- regr.h2o.gbm
-- regr.h2o.glm
-- regr.h2o.randomForest
````

makes Flo's and my previous pull requests obsolete, can be closed. Created a new pull request on a branch now, while previous ones were on a fork.

### CODE REVIEW

**Comment by berndbischl** [comment]
1) remove ' note = "" ', this is the default
2) check again what learners respect weights

3) lines like this 
 pcol = grepl("p[[:digit:]]", colnames(p.df))
must be commented what happens here
basically what we need to do: if the class.levels in task.desc are numbers, and the output colnames are 
exactly "p<number>", the remove the p.
this might work
grepl("^p[[:digit:]]+$", "p1111")


4) 
"'distribution' is set automatically by mlr."
the "by mlr" we dont need, but that it is set to GAUSSIAN is relevant.
recheck all notes!

at the end, rebase on master

**Comment by arnocandel** [comment]
-- regr.h20.randomForest
should be
-- regr.h2o.randomForest
:)

**Comment by arnocandel** [comment]
Also, -- regr.h2o.deeplearning should exist as well, it can do lots of cool distributions (gaussian, gamma, poisson, quantile, tweedie, laplace, etc, incl. offsets).

**Comment by berndbischl** [comment]
@arnocandel 
thx!

@Coorsaa 
Can you pls adapt?

**Comment by Coorsaa** [comment]
@berndbischl and @arnocandel 
Thx for your remakrs, tried to fix them and additionally implemented the regr.h2o.deeplearning learner.

File: R/RLearner_classif_h2odeeplearning.R [hunk 1]
```diff
@@ -0,0 +1,270 @@
...
+      makeNumericLearnerParam("epsilon", default = 1e-08, lower = 1e-10, upper = 1e-4),
+      makeNumericLearnerParam("rate", default = 0.005, lower = 0, upper = 1),
+      makeNumericLearnerParam("rate_annealing", default = 1e-06, lower = 0),
+      makeNumericLearnerParam("rate_decay", default = 1, lower = 0),
+      makeNumericLearnerParam("momentum_start", default = 0),
+      makeNumericLearnerParam("momentum_ramp", default = 1e+06),
+      makeNumericLearnerParam("momentum_stable", default = 0),
+      makeLogicalLearnerParam("nesterov_accelerated_gradient", default = TRUE),
+      makeNumericLearnerParam("input_dropout_ratio", default = 0),
+      makeNumericLearnerParam("hidden_dropout_ratios", default = 0.5),
```
Comment (giuseppec) [inline R/RLearner_classif_h2odeeplearning.R]: Please note that `hidden_dropout_ratios` could also be a vector of the same length as the `hidden` parameter to allow a different dropout ratio in each hidden layer, see e.g. http://didericksen.github.io/deeplearning-r-h2o/.

**Comment by j-hartshorn** [comment]
Hi, can I ask why this learner is limited to Gaussian?

**Comment by larskotthoff** [comment]
You can add the respective parameters and settings and submit a pull request.

**Comment by j-hartshorn** [comment]
@larskotthoff Ok, I'll give it a go. Do you know why it was limited in the first place?

**Comment by larskotthoff** [comment]
Not sure -- other distributions are actually allowed for regr.h2o.deeplearning. It could be that only Gaussian worked at the time, or other values don't make sense for the other learners.

**Pull request merged**